client/dashboard/sites/logs/dataviews/fields.tsx

```diff
@@ -0,0 +1,320 @@
+import { useSuspenseQuery, useQuery } from '@tanstack/react-query';
+import { useMemo } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { useLocale } from '../../../app/locale';
+import { siteBySlugQuery } from '../../../app/queries/site';
+import { siteSettingsQuery } from '../../../app/queries/site-settings';
+import { siteRoute } from '../../../app/router';
+import { LogType } from '../../../data/site-logs';
+import { formatDateWithOffset, getUtcOffsetDisplay } from '../../../utils/datetime';
+import type { PHPLog, ServerLog } from '../../../data/site-logs';
+import type { Field, Operator } from '@wordpress/dataviews';
+
+import './style.scss';
+
+const VALUES_CACHED = [ 'false', 'true' ] as const;
+const VALUES_RENDERER = [ 'php', 'static' ] as const;
+const VALUES_REQUEST_TYPE = [ 'GET', 'HEAD', 'POST', 'PUT', 'DELETE' ] as const;
+const VALUES_SEVERITY = [ 'User', 'Warning', 'Deprecated', 'Fatal error' ] as const;
+const VALUES_STATUS = [ '200', '301', '302', '400', '401', '403', '404', '429', '500' ] as const;
+
+const getLabelCached = ( cached: string ) => {
+	switch ( cached ) {
+		case 'false':
+			return 'False';
+		case 'true':
+			return 'True';
+		default:
+			return cached;
+	}
+};
+const getLabelRenderer = ( renderer: string ) => {
+	switch ( renderer ) {
+		case 'php':
+			return 'PHP';
+		case 'static':
+			return 'Static';
+		default:
+			return renderer;
+	}
+};
+
+const toSeverityClass = ( severity: PHPLog[ 'severity' ] ) =>
+	severity.split( ' ' )[ 0 ].toLowerCase();
+
+export function useFields( { logType }: { logType: LogType } ): Field< PHPLog | ServerLog >[] {
+	const { siteSlug } = siteRoute.useParams();
+	const { data: site } = useSuspenseQuery( siteBySlugQuery( siteSlug ) );
+	const siteId = site?.ID as number;
+
+	const { data: gmtOffset = 0 } = useQuery( {
+		...siteSettingsQuery( siteId ),
+		select: ( siteSettings ) => siteSettings?.gmt_offset ?? 0,
+	} );
+
+	const locale = useLocale();
+
+	const dateTimeLabel = sprintf(
+		/* Translators: %s is the site's GMT offset */
+		__( 'Date & time (%s)' ),
+		getUtcOffsetDisplay( gmtOffset )
+	);
+
+	return useMemo( () => {
+		if ( logType === LogType.PHP ) {
+			return [
+				{
+					id: 'timestamp',
+					type: 'date',
+					label: dateTimeLabel,
+					enableHiding: false,
+					enableSorting: true,
+					getValue: ( { item } ) => ( item as PHPLog ).timestamp,
+					render: ( { field, item } ) => (
+						<span>
+							{ ( field.getValue( { item } ) as string )
+								? formatDateWithOffset( field.getValue( { item } ) as string, gmtOffset, locale )
+								: '' }
+						</span>
+					),
+				},
+				{
+					id: 'severity',
+					type: 'text',
+					label: __( 'Severity' ),
+					enableSorting: false,
+					elements: VALUES_SEVERITY.map( ( severity ) => ( { value: severity, label: severity } ) ),
+					getValue: ( { item } ) => ( item as PHPLog ).severity,
+					render: ( { field, item } ) => (
+						<Badge
```

Comment:
I left a note about the usage of the Badge component in the PR description. Open to changing this.